bndtools.core/src/org/bndtools/core/ui/wizards/service/NewBndServiceWizardPageOne.java

```diff
@@ -0,0 +1,365 @@
+package org.bndtools.core.ui.wizards.service;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.bndtools.api.BndtoolsConstants;
+import org.bndtools.api.ProjectPaths;
+import org.bndtools.templating.Template;
+import org.bndtools.utils.jface.ProgressRunner;
+import org.eclipse.core.filesystem.URIUtil;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.jdt.core.IClasspathAttribute;
+import org.eclipse.jdt.core.IClasspathEntry;
+import org.eclipse.jdt.core.JavaCore;
+import org.eclipse.jdt.internal.core.ClasspathEntry;
+import org.eclipse.jdt.ui.wizards.NewJavaProjectWizardPageOne;
+import org.eclipse.jface.dialogs.ErrorDialog;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.ui.IWorkingSet;
+import org.osgi.service.metatype.AttributeDefinition;
+import org.osgi.service.metatype.ObjectClassDefinition;
+
+import aQute.bnd.build.Workspace;
+import aQute.bnd.exceptions.Exceptions;
+import bndtools.Plugin;
+import bndtools.central.Central;
+import bndtools.wizards.project.ProjectTemplateParam;
+
+public class NewBndServiceWizardPageOne extends NewJavaProjectWizardPageOne {
+
+	protected final ServiceApiProjectNameGroup		nameGroup		= new ServiceApiProjectNameGroup();
+	protected final ServiceApiProjectLocationGroup	locationGroup	= new ServiceApiProjectLocationGroup("Location");
+
+	protected final NewBndServiceWizardPageOne		parentPageOne;
+	protected final String							projectNameSuffix;
+	protected final String							templateName;
+
+	/**
+	 * Constructor used with built-in/explicitly-provided templates
+	 */
+	public NewBndServiceWizardPageOne() {
+		this(null);
+	}
+
+	/**
+	 * Construtor used with named template (type=service-api,service-impl, and
+	 * service-consumer)
+	 *
+	 * @param templateName the name of the template to look for. May be null if
+	 *            no template to be found/loaded.
+	 */
+	public NewBndServiceWizardPageOne(String templateName) {
+		this(null, null, templateName);
+	}
+
+	public NewBndServiceWizardPageOne(NewBndServiceWizardPageOne parentPageOne, String projectNameSuffix,
```

Comment:
I didn't look at the details. However, I think it is better to set the bnd model and then from the bndfile generate this information in one place. See WorkspaceSynchronizer.createProject() is where I tried to standardize this for bndtools. My goal has always been to split any functionality in 2 parts. One thing that can be done in bnd generically (which tends to be a lot easier to test and write) and then small mappings to the different drivers like Gradle, bnd command line, Maven, Eclipse, Intellij,, and VSCode whenever I get around. However, this has been an up hill battle :-(

Reply:
I'm still confused.   Ultimately this new service wizard creates three projects.  Won't it be sort of strange for the user if they run this wizard multiple times and it resets workspace-level values each time?  I 'm still confused.

Reply:
No, the other way around ... you should _follow_ what is in the workspace, never ever set it. I hope the other explanations made it more clear now.

Reply:
Yeah I didn't think that it would make sense to set it at the workspace level as that work be terrible:  user says use javalevel 17 for this project in creation wizard and it changes things to Java-17 in the workspace...breaking all other projects that are there (and expect to stay at Java 16).

So how about this:   How can I query the bndtools workspace javaLevel?   If I can do that in wizard code, I'll leave out the javaLevel if the project wizard selected level is same as workspace.   If' it's not same, then I'll put in new project bnd.bnd.

Actually, if you will tell me how I can query the src and bin/target paths currently set in workspace I'll do that for these settings as well.   I think that will work well...if users are using your approach (workspace rules all) then I won't pollute the meta-data.  If they are not then then their wishes expressed through wizard UI won't be ignored.